pkg/controller/machine_bootstrap_token.go

```diff
@@ -56,41 +58,61 @@ func (c *controller) addBootstrapTokenToUserData(machineName string, driver driv
 	return nil
 }
 
-func (c *controller) getBootstrapTokenOrCreateIfNotExist(machineName string) (secret *v1.Secret, err error) {
-	tokenID, secretName := getTokenIDAndSecretName(machineName)
-
-	secret, err = c.targetCoreClient.CoreV1().Secrets(metav1.NamespaceSystem).Get(secretName, metav1.GetOptions{})
+func (c *controller) getBootstrapTokenOrCreateIfNotExist(machineLabels map[string]string, machineName string) (secret *v1.Secret, err error) {
+	value, ok := machineLabels["instance-group"]
+	if !ok || value == "" {
+		return nil, fmt.Errorf("missing instance-group label from machine: '%v'", machineName)
```

Comment:
let add fall back to old creating secret if we have errors and log with warning in this cases